### DISCUSSION TITLE
Welcome to streamlit-pdf-viewer Discussions!

### DISCUSSION THREAD

**Discussion by lfoppiano** [lfoppiano/streamlit-pdf-viewer#59] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Naveenredie15** [comment]
I'm trying to create an expander for each page in a PDF, but I'm unable to achieve this using the following code. Can you help me?
for page in range(1, get_page_count(ss.pdf_ref.getvalue())+1):
    page_expander = st.expander(f"Page {page}", expanded=False)
    page_expander.write(pdf_viewer(
        input=ss.bytes_data, pages_to_render=[1], key=f"pdf_viewer_{page}"
    ))

**Reply by lfoppiano** [reply]
What is the issue? 

have you tried adding the `width` when you call the pdfviewer?

**Reply by lfoppiano** [reply]
you cannot write the pdf_viewer in the expander, maybe you can try with

```
with st.expander....: 
     pdf_viewer(.....)

**Comment by drnihalmehta** [comment]
Great component. However, the scroll_to_page doesn't work when height is specified. 
Is there any way to scroll to a page when height and width are specified?
Thank you.
Nihal

**Reply by lfoppiano** [reply]
hi @drnihalmehta, can you share me the code you're using where the scroll_to_page does not work? 

Alternatively you can avoid specifying the height and wrap the component into a `st.container`: 

```
with st.container(height=xyz):
   pdf_viewer(....
```

**Reply by lfoppiano** [reply]
@drnihalmehta is it working?

**Comment by drnihalmehta** [comment]
Hi Luca,
I’ll try wrapping within container.

Here’s my test code:
import streamlit as st from streamlit_pdf_viewer import pdf_viewer # Path
to the PDF file folder = "pdfs" filename =
"77_yr_breathing_swallowing_difficulty.pdf" pdf_path =
f"{folder}/{filename}" # Streamlit app title st.title("PDF Viewer - Using
Binary Data") st.write(f"{pdf_path}") # Read the PDF as binary data try:
  with open(pdf_path, "rb") as pdf_file:         pdf_bytes =
pdf_file.read()     # Pass the binary data directly to pdf_viewer     #
pdf_viewer(pdf_bytes, scroll_to_page=3, width=700, height=700) except
FileNotFoundError:     st.error(f"File not found: {pdf_path}")


Thanks and best regards, Nihal

**Reply by lfoppiano** [reply]
Sorry can't you please take some time to format it properly? I can't read it

**Comment by drnihalmehta** [comment]
Ok. I’ll send you the .py file itself shortly where it’s
properly formatted.

I cut and pasted into the email and it’s showing up nicely on my end.

Sorry about that.

Thanks and best regards, Nihal

**Comment by drnihalmehta** [comment]
Here's the sample test file I'm using.

Also, it doesn't work when using a URL for the location of the PDF
location.

Both cases are in this file.

Thanks for your help.

Thanks and best regards,
Nihal

Nihal J. Mehta, Ph.D.
***@***.***

**Reply by lfoppiano** [reply]
URL location is not supported, it's well explained in the documentation. 

I could not see the code. Please paste it with markdown directly in github instead of answering by email.

**Comment by drnihalmehta** [comment]
Here is the code. Thanks again.
```
import streamlit as st
from streamlit_pdf_viewer import pdf_viewer

# Path to the PDF file
folder = "pdfs"
filename = "test.pdf"
pdf_path = f"{folder}/{filename}"

# Streamlit app title
st.title("PDF Viewer - Using Binary Data")
st.write(f"{pdf_path}")
# Read the PDF as binary data
try:
    with open(pdf_path, "rb") as pdf_file:
        pdf_bytes = pdf_file.read()
        pdf_viewer(pdf_bytes, scroll_to_page=3)
  
except FileNotFoundError:
    st.error(f"File not found: {pdf_path}")



```

**Comment by drnihalmehta** [comment]
I got it to work using st.container as you suggested.

Thanks and best regards, Nihal

**Comment by FrancescoBoschi** [comment]
Hi, I am trying to render a pdf and scroll to a specific page.
The pdf is shown properly but the scroll action is not happening.
This is my code:

```
if st.session_state['references']:
        for i, source in enumerate(st.session_state['references']):
            if st.button(f"{source['file_name']}"):
                # Fetch and display the PDF
                #
                modal = Modal(key = "modal key", title=f"""{source['file_name']}
                Page of Interest: {source['page']}""" , max_width=1000)
                with modal.container():
                    pdf_viewer(get_signed_url(source["url"], url, headers), width=1000, height=1000, scroll_to_page=int(source["page"]))
```

I also tried without Modal, but it's still the same.
I noticed that if I resize the browser Windows it actually scrolls down to the desired page.
How can I fix it?

**Reply by lfoppiano** [reply]
It's probably a bug, we need to investigate, if you use version 0.0.18 it should work, see another report here #83
